app/main/views/unsubscribe_requests.py

```diff
@@ -41,3 +43,59 @@ def unsubscribe_request_report(service_id, batch_id=None):
         form=form,
         error_summary_enabled=True,
     )
+
+
+@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/download")
+@main.route("/services/<uuid:service_id>/unsubscribe-requests/reports/download/<uuid:batch_id>.csv")
+@user_has_permissions("view_activity")
+def download_unsubscribe_request_report(service_id, batch_id=None):
+    if not batch_id:
+        return redirect(url_for("main.create_unsubscribe_request_report", service_id=service_id))
+    else:
+        try:
+            report = service_api_client.get_unsubscribe_request_report(service_id, batch_id)
+            column_names = {
+                "email_address": "Email address",
+                "template_name": "Template name",
+                "original_file_name": "Contact list file name",
+                "template_sent_at": "Template sent at",
+            }
+            # initialise with header row
+            data = [list(column_names.values())]
+            for row in report["unsubscribe_requests"]:
+                data.append([row[key] for key in column_names.keys()])
+            return (
+                Spreadsheet.from_rows(data).as_csv_data,
+                200,
+                {
+                    "Content-Type": "text/csv; charset=utf-8",
+                    "Content-Disposition": f'attachment; filename="Email unsubscribe requests '
+                    f"{format_date_normal(report['earliest_timestamp'])} "
+                    f"to {format_date_normal(report['latest_timestamp'])}.csv",
```

Comment:
Filenames should have numeric dates so they are listed in order once downloaded
```suggestion
                    f"{format_date_numeric(report['earliest_timestamp'])} "
                    f"to {format_date_numeric(report['latest_timestamp'])}.csv",
```